### PR TITLE
Python3.6 depreciations

### PR DESCRIPTION
**Pull request by jgarst**
Python 3.6 has depreciated a module tkinter.tix, the last remaining importlib.load_module occurrences, and dropped inspect.getargspec altogether.  

https://docs.python.org/3.6/whatsnew/3.6.html#deprecated-python-modules-functions-and-methods

This pull request, adds the depreciated methods and module checks, and separates depreciated function testing into python3.5- and python3.6+.

Please let me know if there is anything else you need from these changes.
-Jared

### CODE REVIEW

File: pylint/test/functional/deprecated_methods_py36.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+""" Functional tests for method deprecation. """
```
Comment (PCManticore) [inline pylint/test/functional/deprecated_methods_py36.py]: Ideally, this file should contain only the code for the deprecations happening in the same Python version, which means, in this case you should test only for the deprecation of FileLoader.load_module.

Comment (jgarst) [inline pylint/test/functional/deprecated_methods_py36.py]: I did a full copy of the tests to account for inspect.getmoduleinfo being removed from 3.6. 
The two options I can see are splitting the inspect.getmoduleinfo line into it's own version dependent test (changing the old test), or moving to a per version list of everything that is depreciated (duplicating tests).

If you confirm that you like the first version better, I'll update the pull request.

Comment (PCManticore) [inline pylint/test/functional/deprecated_methods_py36.py]: Yep, I would like the first one more.

**Comment by PCManticore** [comment]
Merged locally: https://github.com/PyCQA/pylint/commit/3a83a093d14d59d5026c0beae6bf025fe6b4ded7

**Pull request closed without merging**